docs: add Electron 30 blog post (#553)
electron/website: blog/electron-30-0.md

  - See [our Web Embeds documentation](https://www.electronjs.org/docs/latest/tutorial/web-embeds) for a comparison of the new `WebContentsView` API to other similar APIs.
- Implemented support for the [File System API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_API) ([#41827](https://github.com/electron/electron/commit/cf1087badd437906f280373decb923733a8523e6))

### Stack Changes

- Chromium `124.0.6367.49`
  - New in [Chrome 124](https://developer.chrome.com/blog/new-in-chrome-124/) and in [DevTools 124](https://developer.chrome.com/blog/new-in-devtools-124/)
  - New in [Chrome 123](https://developer.chrome.com/blog/new-in-chrome-123/) and in [DevTools 123](https://developer.chrome.com/blog/new-in-devtools-123/)
- Node `20.11.1`
  - [Node 20.11.1 notes](https://nodejs.org/en/blog/release/v20.11.1/)
- V8 `12.4`

Electron 30 upgrades Chromium from `122.0.6261.39` to `124.0.6367.49`, Node from `20.9.0` to `20.11.1`, and V8 from `12.2` to `12.4`.

### New Features

- Added a `transparent` webpreference to webviews. ([#40301](https://github.com/electron/electron/pull/40301))

> - Added a new instance property `navigationHistory` on webContents API with `navigationHistory.getEntryAtIndex` method, enabling applications to retrieve the URL and title of any navigation entry within the browsing history. ([#41662](https://github.com/electron/electron/pull/41662)) <span style="font-size:small;">(Also in [29](https://github.com/electron/electron/pull/41661))</span>

Author: It looks like this release note is pretty accurate to how the API is meant to be used. It's a little bit unclear given the new `navigationHistory` module that exposes a class that can't be instantiated by devs, but the description in the release note is the intended use of this new feature by devs.

- Added a new method `BrowserWindow.isOccluded()` to allow apps to check occlusion status. ([#38982](https://github.com/electron/electron/pull/38982))

> - Added proxy configuring support for requests made with net module from utility process. ([#41417](https://github.com/electron/electron/pull/41417)) <span style="font-size:small;">(Also in [28](https://github.com/electron/electron/pull/41744), [29](https://github.com/electron/electron/pull/41416))</span>

Reviewer: ```suggestion
- Added proxy configuring support for requests made with the `net` module from the utility process. ([#41417](https://github.com/electron/electron/pull/41417)) <span style="font-size:small;">(Also in [28](https://github.com/electron/electron/pull/41744), [29](https://github.com/electron/electron/pull/41416))</span>
```

- Added support for Bluetooth ports being requested by service class ID in `navigator.serial`. ([#41734](https://github.com/electron/electron/pull/41734)) <span style="font-size:small;">(Also in [29](https://github.com/electron/electron/pull/41735))</span>

> - Added support for `NODE_EXTRA_CA_CERTS`. ([#41822](https://github.com/electron/electron/pull/41822))

Reviewer: ```suggestion
- Added support for the Node.js [`NODE_EXTRA_CA_CERTS`](https://nodejs.org/api/cli.html#node_extra_ca_certsfile) CLI flag. ([#41822](https://github.com/electron/electron/pull/41822))
```

title: Electron 30.0.0
date: 2024-04-16T00:00:00.000Z
authors:
  - name: clavin
    url: 'https://github.com/clavin'
    image_url: 'https://github.com/clavin.png?size=96'
  - name: VerteDinde
    url: 'https://github.com/vertedinde'
    image_url: 'https://github.com/vertedinde.png?size=96'
slug: electron-30-0
---

Electron 30.0.0 has been released! It includes upgrades to Chromium `124.0.6367.49`, V8 `12.4`, and Node.js `20.11.1`.

---

The Electron team is excited to announce the release of Electron 30.0.0! You can install it with npm via `npm install electron@latest` or download it from our [releases website](https://releases.electronjs.org/releases/stable). Continue reading for details about this release.

If you have any feedback, please share it with us on [Twitter](https://twitter.com/electronjs) or [Mastodon](https://social.lfx.dev/@electronjs), or join our community [Discord](https://discord.com/invite/electronjs)! Bugs and feature requests can be reported in Electron's [issue tracker](https://github.com/electron/electron/issues).

## Notable Changes

### Highlights

- ASAR Integrity fuse now supported on Windows [#40504](https://github.com/electron/electron/pull/40504)

>   - Existing apps with ASAR Integrity enabled may not work on Windows if not configured correctly.

Reviewer: We should give people a one-liner upgrade path without clicking through

```suggestion
  - Existing apps with ASAR Integrity enabled may not work on Windows if not configured correctly. Apps using Electron packaging tools should upgrade to `@electron/packager@18.3.1` or `@electron/forge@7.4.0`.
```

> * ASAR Integrity fuse now supported on Windows
>   * Existing apps with ASAR Integrity enabled may not work on Windows if not configured correctly.
>   * Take a look at our [ASAR Integrity tutorial](https://www.electronjs.org/docs/latest/tutorial/asar-integrity) for more information.
> * Added `WebContentsView` and `BaseWindow`, deprecating & replacing `BrowserView`
> * Added `net` module to utility process
> * Implemented support for the File System API

Reviewer 2: I'd add the PR numbers/links here so people can click through and view them

```suggestion
* ASAR Integrity fuse now supported on Windows [#40504](https://github.com/electron/electron/pull/40504)
  * Existing apps with ASAR Integrity enabled may not work on Windows if not configured correctly.
  * Take a look at our [ASAR Integrity tutorial](https://www.electronjs.org/docs/latest/tutorial/asar-integrity) for more information.
* Added `WebContentsView` and `BaseWindow`, deprecating & replacing `BrowserView` [#35658](https://github.com/electron/electron/pull/35658)
* Added `net` module to utility process [#40017](https://github.com/electron/electron/pull/40017)
* Implemented support for the File System API [#41827](https://github.com/electron/electron/commit/cf1087badd437906f280373decb923733a8523e6)
```

> * Added `WebContentsView` and `BaseWindow`, deprecating & replacing `BrowserView` [#35658](https://github.com/electron/electron/pull/35658)

Reviewer: ```suggestion
* Added [`WebContentsView`](https://www.electronjs.org/docs/latest/api/web-contents-view) and [`BaseWindow`](https://www.electronjs.org/docs/latest/api/base-window) main process modules, deprecating & replacing `BrowserView` [#35658](https://github.com/electron/electron/pull/35658)
```

> - Added `WebContentsView` and `BaseWindow`, deprecating & replacing `BrowserView` [#35658](https://github.com/electron/electron/pull/35658)

Reviewer: We should point to https://www.electronjs.org/docs/latest/tutorial/web-embeds to compare and contrast `WebContentsView` as well.

Reviewer: issue(blocking): I don't think the release notes are clear enough about the BrowserView implementation being a JavaScript shim for WebContentsView now.

Reviewer: I guess "deprecating and replacing" sounds a bit vague to me as in the old impl doesn't exist in the code anymore.

> * Added `net` module to utility process [#40017](https://github.com/electron/electron/pull/40017)

Reviewer: ```suggestion
* Added [`net`](https://www.electronjs.org/docs/latest/api/net) module to utility process [#40017](https://github.com/electron/electron/pull/40017)
```

> * Implemented support for the File System API [#41827](https://github.com/electron/electron/commit/cf1087badd437906f280373decb923733a8523e6)

Reviewer: ```suggestion
* Implemented support for the [File System API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_API) [#41827](https://github.com/electron/electron/commit/cf1087badd437906f280373decb923733a8523e6)
```

### Stack Changes

- Chromium `124.0.6367.49`
  - New in [Chrome 124](https://developer.chrome.com/blog/new-in-chrome-124/) and in [DevTools 124](https://developer.chrome.com/blog/new-in-devtools-124/)
  - New in [Chrome 123](https://developer.chrome.com/blog/new-in-chrome-123/) and in [DevTools 123](https://developer.chrome.com/blog/new-in-devtools-123/)
- Node `20.11.1`
  - [Node 20.11.1 notes](https://nodejs.org/en/blog/release/v20.11.1/)
- V8 `12.4`

Electron 29 upgrades Chromium from `122.0.6261.39` to `124.0.6367.49`, Node from `20.9.0` to `20.11.1`, and V8 from `12.2` to `12.4`.

### Breaking Changes

#### Behavior Changed: cross-origin iframes now use Permission Policy to access features

Cross-origin iframes must now specify features available to a given `iframe` via the `allow`
attribute in order to access them.

See [documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#allow) for
more information.

> #### Removed: The `--disable-color-correct-rendering` switch

Reviewer: nit: wording

```suggestion
#### Removed: The `--disable-color-correct-rendering` command line switch
```

This switch was never formally documented but it's removal is being noted here regardless. Chromium itself now has better support for color spaces so this flag should not be needed.

#### Behavior Changed: `BrowserView.setAutoResize` behavior on macOS

> In Electron 30, BrowserView is now a wrapper around the new [WebContentsView](api/web-contents-view.md) API.

Reviewer 2: I think lint is failing here - we can add the full link:
```suggestion
In Electron 30, BrowserView is now a wrapper around the new [WebContentsView](https://www.electronjs.org/docs/latest/api/web-contents) API.
```

Reviewer 3: This is still failing due to the current SNAFU with deploying the latest docs to the website - it's a legit 404.

> This switch was never formally documented but it's removal is being noted here regardless. Chromium itself now has better support for color spaces so this flag should not be needed.

Reviewer: nit: grammar

```suggestion
This switch was never formally documented but its removal is being noted here regardless. Chromium itself now has better support for color spaces so this flag should not be needed.
```
